### PR TITLE
esp32,esp8266: Update espnow docs for WLAN.config(pm=x) option.

### PR DESCRIPTION
**Pull request by glenn20**
Updates for the espnow module documentation following merge of #8993.

Also includes a few updates and clarifications following merge of #6515.

Update docs/library/espnow.rst to add:
- guidance on using WLAN.config(pm=WLAN.PM_NONE) for reliable espnow performance while also connected to a wifi access point.
- correction on use of add_peer() for receiving encrypted messages
- correction for default value of "encrypt" parameter (add_peer())
- guidance on use of ESPNow.irq(): recommand users readout all messages in the buffer each time the callback is called.

### CODE REVIEW

File: docs/library/espnow.rst [hunk 1]
```diff
@@ -739,7 +753,9 @@ point. When an ESP32 or ESP8266 device connects to a Wifi Access Point (see
 `ESP32 Quickref <../esp32/quickref.html#networking>`__) the following things
 happen which affect ESPNow communications:
 
-1. Wifi Power-saving Mode is automatically activated and
+1. Wifi Power-saving Mode (
+   `sta.config(pm=sta.PM_PERFORMANCE<network.WLAN.PM_PERFORMANCE>`)
```
Comment (dpgeorge) [inline docs/library/espnow.rst:757]: I'm not sure what's going on here, but it should be either `sta.PM_PERFORMANCE` or `network.WLAN.PM_PERFORMANCE`.

Note that 53cb073571196d2ba7174706d5d784e42e5430cf was merged so the latter will work.

File: docs/library/espnow.rst [hunk 2]
```diff
@@ -780,6 +799,7 @@ espnow:
   sta.connect('myssid', 'mypassword')
   while not sta.isconnected():  # Wait until connected...
       time.sleep(0.1)
+  sta.config(pm=sta.PM_NONE)  # ..then disable power saving
   ap.active(True)         # Disable power-saving mode
```
Comment (dpgeorge) [inline docs/library/espnow.rst:803]: Should this line `ap.active(True)` be removed now that the STA can disable power saving?

File: docs/library/espnow.rst [hunk 3]
```diff
@@ -750,17 +765,21 @@ device to turn off the radio periodically (typically for hundreds of
...
    - You **may** also choose to send your messages via the AP_IF interface, but
      this is not necessary.
    - ESP8266 peers must send messages to this AP_IF interface (see below).
 
-2. Configuring ESPNow clients to retry sending messages.
+3. Configuring ESPNow clients to retry sending messages.
 
 **Receiving messages from an ESP8266 device:** Strangely, an ESP32 device
-connected to a wifi network using method 1 or 2 above, will receive ESP-Now
+connected to a wifi network using method 1 or 2 above, will receive ESP-NOW
```
Comment (dpgeorge) [inline docs/library/espnow.rst:782]: In most other places here it's written as `ESPNow`.  Why is this one different?

File: docs/library/espnow.rst [hunk 4]
```diff
@@ -383,9 +383,15 @@ after reboot/reset). This reduces the reliability of receiving ESP-NOW messages
 Peer Management
 ---------------
 
-The Espressif ESP-Now software requires that other devices (peers) must be
-*registered* before we can `send()<ESPNow.send()>` them messages. It is
-**not** necessary to *register* a peer to receive a message from that peer.
+On ESP32 devices, the Espressif ESP-Now software requires that other devices
```
[the remark this answers is not in the document]
Comment (glenn20) [reply docs/library/espnow.rst:386]: I will review and fix up consistency throughout the doc. Espressif calls the
protocol ESP-NOW. Micropython supports this through the ESPNow python class
API.

**Comment by dpgeorge** [comment]
Thanks for updating.

**Pull request merged**